internal/provider/resources/user/user.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package user
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	tsbv2 "github.com/tetrateio/api/tsb/v2"
+
+	"github.com/tetratelabs/terraform-provider-tsb/internal/helpers"
+	"github.com/tetratelabs/terraform-provider-tsb/internal/provider/defaults"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &UserResource{}
+var _ resource.ResourceWithImportState = &UserResource{}
+
+func NewResource() resource.Resource {
+	return &UserResource{}
+}
+
+// UserResource defines the resource implementation.
+type UserResource struct {
+	client tsbv2.TeamsClient
+}
+
+type userResourceModel struct {
+	Id           types.String `tfsdk:"id"`
+	Name         types.String `tfsdk:"name"`
+	Organization types.String `tfsdk:"organization"`
+	DisplayName  types.String `tfsdk:"display_name"`
+	LoginName    types.String `tfsdk:"login_name"`
+	FirstName    types.String `tfsdk:"first_name"`
+	LastName     types.String `tfsdk:"last_name"`
+	Email        types.String `tfsdk:"email"`
+	SourceType   types.String `tfsdk:"source_type"`
+}
+
+func (r *UserResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_user"
+}
+
+// Schema implements resource.Resource
+func (*UserResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
```

Comment:
Is this (or is this goind to be) autogenerated?

Reply:
It is mostly autogenerated. It's a copy/paste/change slightly deal.

Reply:
Copy paste from the auto-generated code.